### DISCUSSION TITLE
User Interface - General Thoughts

### DISCUSSION THREAD

**Discussion by ninapaley** [mysticsymbolic/mysticsymbolic.github.io#156] [Ideas]
Thoughts on UX:

**MysticSymbolic is an art generator, not a graphics editor or animation program.**
Its primary purpose is to generate art.
All of its current UX features (color sliders, checkboxes) are to help us develop the art generator. Most will be absent in the final version.

If users want to deeply edit images and animation, they can copy and drop the assets into any number of existing image editors and animation programs. We do not need to re-invent those wheels.

Some minor editing capabilities would be a nice feature, but aren’t essential and aren’t a priority right now. If/when the time comes, I suggest the ability to right-click a symbol for a dropdown menu with the following options:

- scale
- rotate (around centroid)
- move
- replace with different symbol (second dropdown menu)
- replace with different symbol (random)
- change fill color (color picker)
- change stroke color (color picker)
- change animation (see #124 )
- delete/remove

(These changes would apply only to the individual symbol in question. Everything else would stay the same (color, scale, orientation, etc.). All of the connectivity determined by a symbol’s specs would be broken the moment an individual symbol is modified; at that point, a symbol’s specs become irrelevant, and it’s just a shape.)

**Comment by hemulionbotanisti** [comment]
I have an idea! It would be nice to have a "go back" button in the interface for going back (maybe 1-2 generated clusters) to the previous cluster. I tend to click the randomize button too fast for my brain to catch the desired cluster and I frequently wish I could go back to the previous cluster! :D

**Comment by toolness** [comment]
Oh this is actually already supported @hemulionbotanisti!  You just use your browser's "back" button, there's no "back" UI in the tool itself :)

**Reply by hemulionbotanisti** [reply]
Oh you're right! I remember trying this some time ago and the browser just went to the previous web page. Now it works just as you described, thank you! 🙏
